app/models/package_manager/maven.rb

```diff
@@ -158,10 +158,14 @@ def self.dependencies(name, version, project)
       end
     end
 
-    def self.versions(_project, name)
-      xml_metadata = get_raw(MavenUrl.from_name(name, repository_base, NAME_DELIMITER).maven_metadata)
-      xml_versions = Nokogiri::XML(xml_metadata).css("version").map(&:text)
-      retrieve_versions(xml_versions.filter { |item| !item.ends_with?("-SNAPSHOT") }, name)
+    def self.versions(project, name)
+      if project && project[:versions]
+        return project[:versions]
+      else
+        xml_metadata = get_raw(MavenUrl.from_name(name, repository_base, NAME_DELIMITER).maven_metadata)
+        xml_versions = Nokogiri::XML(xml_metadata).css("version").map(&:text)
+        retrieve_versions(xml_versions.filter { |item| !item.ends_with?("-SNAPSHOT") }, name)
+      end
```

Comment:
@jsonperl turns out this double-fetch was mostly a problem with Maven, so this uses the pre-fetched `project` var if it's available, instead of refetching the versions.

Reply:
(this also only speeds up the `sync_version: :all` case, and I'm going to work on speeding up the single-version update after this PR)